src/genpy/dynamic.py

```diff
@@ -111,13 +111,15 @@ def _gen_dyn_modify_references(py_text, current_type, types):
     return py_text
 
 
-def generate_dynamic(core_type, msg_cat):
+def generate_dynamic(core_type, msg_cat, tmp_dir_path=None, tmp_file_name=None):
```

Comment:
Are you planning on passing only `tmp_dir_path` but not `tmp_file_name`?

If not I would argue this could be reduced to a single optional argument which contains the full path to the file.

Reply:
i agree. i reduced it a single optional argument which contains the full path to the file.